src/main/java/prefs/PanelInfo.java

```diff
@@ -23,4 +23,21 @@ public String getPanelFilter() {
         return this.filter;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        PanelInfo panelInfo = (PanelInfo) o;
+
+        if (name != null ? !name.equals(panelInfo.name) : panelInfo.name != null) return false;
+        return filter != null ? filter.equals(panelInfo.filter) : panelInfo.filter == null;
+    }
```

Comment:
The 2 lines above are not every readable, especially using the result of a `?` operator in the conditional expression for the `if` statement. You can use 2 `||` operators instead. But it's better to have multiple if clauses or expressions to reflect clearly the equality criteria.